src/Core/tests/UnitTests/Hosting/RegisteredHandlerServiceTypeSetTests.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using Microsoft.Maui.Hosting.Internal;
+using Microsoft.Maui.Platform;
+using Microsoft.Maui.UnitTests;
+using Xunit;
+
+namespace Microsoft.Maui.UnitTesting.Hosting
+{
+	[Category(TestCategory.Core, TestCategory.Hosting)]
+	public class RegisteredHandlerServiceTypeSetTests
+	{
+		[Fact]
+		public void ResolveCorrespondingHandlerServiceType()
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+			registeredTypes.Add(typeof(ViewStub));
+
+			var type = registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(ViewStub));
+
+			Assert.Equal(typeof(ViewStub), type);
+		}
+
+		[Fact]
+		public void ThrowsWhenNoMatchingHandlerServiceTypeIsRegistered()
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+
+			Assert.Throws<HandlerNotFoundException>(() => registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(ViewStub)));
+		}
+
+		[Theory]
+		[InlineData(typeof(IViewStub))]
+		[InlineData(typeof(IMyBaseViewStub))]
+		[InlineData(typeof(IMyDerivedViewStub))]
+		public void ResolvesBaseInterfaceServiceType(Type baseInterfaceType)
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+			registeredTypes.Add(baseInterfaceType);
+
+			var resolvedServiceType = registeredTypes.ResolveVirtualViewToRegisteredHandlerServiceType(typeof(MyDerivedViewStub));
+
+			Assert.NotNull(resolvedServiceType);
+		}
+
+		[Fact]
+		public void ResolvesToMostDerivedBaseInterfaceServiceType()
+		{
+			var registeredTypes = new RegisteredHandlerServiceTypeSet();
+			registeredTypes.Add(typeof(IMyDerivedViewStub));
+			registeredTypes.Add(typeof(IMyBaseViewStub));
+			registeredTypes.Add(typeof(IViewStub));
```

Comment:
Could we turn this into a `[Theory]` and test several permutations (not necessarily all permutations) of the registrations? Just to prove that it doesn't depend on the order of the calls to `Add`.